### PR TITLE
Implemented disabling scroll when modal with curtain is opened

### PR DESCRIPTION
**Pull request by saragajic11**
This PR has been opened because previous 2 (https://github.com/adobecom/milo/pull/1457 &
https://github.com/adobecom/milo/pull/1419) had outdated and broken main origin branch. I am opening this PR after cloning this project from the start, with fresh and up-to-date main branch.

This PR should resolve the issue of scrollable background when modal dialog (which is authored to have a 'curtain') is opened. In other cases (with 'curtain-off') background should be scrollable per requirements.

Resolves: [MWPW-136767](https://jira.corp.adobe.com/browse/MWPW-136767)

Test URLs:

Before: https://main--milo--adobecom.hlx.page/drafts/rclayton/video-examples?martech=off
After: https://mwpw-136767-modal-background-scroll--milo--saragajic11.hlx.page/drafts/rclayton/video-examples?martech=off

Commerce dialog (it should work the same as before this PR):

Before: https://main--milo--adobecom.hlx.page/drafts/ruchika/discover1?martech=off
After: https://mwpw-136767-modal-background-scroll--milo--saragajic11.hlx.page/drafts/ruchika/discover1?martech=off

Example of page which had (in previous PR) _content_ issues:
https://main--cc--adobecom.hlx.page/creativecloud/animation/discover?milolibs=mwpw-136767-modal-background-scroll--milo--saragajic11

### CODE REVIEW

**Review by fullcolorcoder — APPROVED** [review]

**Comment by narcis-radu** [comment]
Hey @saragajic11, can you please update the description to reference and close the previous PRs?
* https://github.com/adobecom/milo/pull/1457
* https://github.com/adobecom/milo/pull/1419

**Review by narcis-radu — CHANGES_REQUESTED** [review]

File: libs/blocks/modal/modal.js [hunk 1]
```diff
@@ -48,6 +47,9 @@ function closeModal(modal) {
     if (mod.classList.contains('dialog-modal')) {
       mod.remove();
     }
+    if (!document.querySelectorAll('.modal-curtain').length) {
```
[the remark this answers is not in the document]
Comment (saragajic11) [reply libs/blocks/modal/modal.js]: good catch, updated: https://github.com/adobecom/milo/pull/1466/commits/f99ec9e00a1cfec161fa4baef1380562f81ecdc3 thanks! @narcis-radu

**Comment by Snehayt** [comment]
Verified the flow in all the browsers(firefox,brave,chrome and edge) in desktop,mac and devices.Works as expected.

image
image
image

**Review by narcis-radu — APPROVED** [review]

**Comment by narcis-radu** [comment]
Looks good 👍 ! You might want to add some unit tests.

**Review by Ruchika4 — APPROVED** [review]

**Pull request merged**